src/vm.rs

```diff
@@ -90,6 +91,42 @@ pub(crate) enum VmExecutionResult {
     },
 }
 
+// https://docs.soliditylang.org/en/v0.8.26/internals/layout_in_storage.html#mappings-and-dynamic-arrays
+fn get_erc20_balance_slot(address: Address) -> U256 {
+    let mut buf = [0u8; 64];
+    buf[12..32].copy_from_slice(address.as_slice());
+    keccak256(buf).into()
+}
+
+#[derive(Debug)]
+enum LazyStrategy {
+    None,
+    RawTransfer,
+    ERC20Transfer {
+        sender_balance_slot: U256,
```

Comment:
Ideally we can simplify this as two 256-bit hashes are fairly large. This would make even a `LazyStrategy::None` 72 bytes in the stack.

Reply:
how the definition is `ERC20Transfer { amount: U256 },`